packages/back-end/src/queryRunners/QueryRunner.ts

```diff
@@ -191,13 +213,77 @@ export abstract class QueryRunner<
     });
   }
 
+  public async startReadyQueries(): Promise<void> {
+    const queuedQueries = await getQueriesByIds(
+      this.model.organization,
+      this.model.queries
+        .filter((q) => q.status === "queued")
+        .map((q) => q.query)
+    );
+    logger.debug(
+      `Starting any queued queries that are ready: ${queuedQueries.map(
+        (q) => q.id
+      )}`
+    );
+    queuedQueries.map(async (query) => {
+      // check if all dependencies are finished
+      // assumes all dependencies are within the model; if any are not, query will hang
+      // in queued state
+
+      const failedDependencies: QueryPointer[] = [];
+      const succeededDependencies: QueryPointer[] = [];
+      const pendingDependencies: QueryPointer[] = [];
+
+      const dependencyIds: string[] = query.dependencies ?? [];
+      dependencyIds.forEach((dependencyId) => {
+        const dependencyQuery = this.model.queries.find(
+          (q) => q.query == dependencyId
+        );
+        if (dependencyQuery === undefined) {
+          throw new Error(`Dependency ${dependencyId} not found in model`);
+        } else if (dependencyQuery.status === "succeeded") {
+          succeededDependencies.push(dependencyQuery);
+        } else if (dependencyQuery.status === "failed") {
+          failedDependencies.push(dependencyQuery);
+        } else {
+          pendingDependencies.push(dependencyQuery);
+        }
+      });
+
+      if (failedDependencies.length) {
+        logger.debug(`${query.id}: Dependency failed...`);
+        updateQuery(query, {
+          finishedAt: new Date(),
+          status: "failed",
+          error: `Dependencies failed: ${failedDependencies.map(
+            (q) => q.query
+          )}`,
+        });
```

Comment:
We should await this call too and do `onQueryFinish()` once we're done updating.  Otherwise, it could end up in a stuck state.